MultiServer.py

```diff
@@ -322,10 +334,14 @@ def _init_game_data(self):
             self.location_names[game].update(archipelago_location_names)
 
     def item_names_for_game(self, game: str) -> typing.Optional[typing.Dict[str, int]]:
-        return self.gamespackage[game]["item_name_to_id"] if game in self.gamespackage else None
+        return (self.gamespackage[game]["item_name_to_id"] |
+                self.dynamic_package.get(game, {"item_name_to_id": {}})["item_name_to_id"]) \
+            if game in self.gamespackage else None
 
     def location_names_for_game(self, game: str) -> typing.Optional[typing.Dict[str, int]]:
-        return self.gamespackage[game]["location_name_to_id"] if game in self.gamespackage else None
+        return (self.gamespackage[game]["location_name_to_id"] |
+                self.dynamic_package.get(game, {"location_name_to_id": {}})["location_name_to_id"]) \
+            if game in self.gamespackage else None
```

Comment:
This turns these methods from barely more than some pointer dereffing into creating a fresh set on each call, which is significantly more performance requirement; and it's used in plenty of places. Rather than creating this union set every time it might be better to split the other code into two lookups or caching this. But the cache might be too memory heavy.

So my proposal is  a new method ala def is_item_name_in_game() and then do a lookup in each set from there.